libs/@local/hash-backend-utils/src/queue/redis.ts

```diff
@@ -178,9 +171,9 @@ export class RedisQueueExclusiveConsumer implements QueueExclusiveConsumer {
     let item = await this.client.rpoplpush(processingName, processingName);
 
     // Otherwise, pop from the main queue and push onto the processing queue.
-    if (!item) {
+    if (item) {
```

Comment:
I don't quite understand why this has swapped. Why do we block if we already have an item?

Reply:
This was not intentional. Thanks. I originally deleted the `queue` abstraction but decided to keep it for now. I don't really think we should keep this implementation around as it seems somewhat error-prone and based on application-level orchestration to work rather than leveraging the data structures provided by redis